sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -0,0 +1,193 @@
+import numpy as np
+from ...utils import check_matplotlib_support
+
+
+def _check_boundary_response_method(estimator, response_method):
```

Comment:
Haven't checked in details but don't we have this logic duplicated in various places, modulo some minor differences?